### DISCUSSION TITLE
Debugging in a client project

### DISCUSSION THREAD

**Discussion by CidTori** [mapstruct/mapstruct#3419] [Q&A]
Hello !

Does anybody know how to debug mapstruct-processor from a project that uses it, on IntelliJ for example?
I tried adding it as a plain dependency to be able to put break points, and running Maven in debug mode, but the break points are ignored 😕

**Comment by filiphr** [comment]
The way I usually debug this in a custom project is by adding the `mapstruct-processor` as a regular dependency, put breakpoints e.g. in the `public boolean process(final Set<? extends TypeElement> annotations, final RoundEnvironment roundEnvironment)` method of the `MappingProcessor`. Then in the Maven Toolbar find your project and in the Lifecycle right click on the compile maven goal and click on debug.
